modules/yieldmoBidAdapter.js

```diff
@@ -1,109 +1,128 @@
+        p: [],
+        page_url: bidderRequest.refererInfo.referer,
+        bust: new Date().getTime().toString(),
+        pr: bidderRequest.refererInfo.referer,
+        scrd: localWindow.devicePixelRatio || 0,
+        dnt: getDNT(),
+        description: getPageDescription(),
+        title: localWindow.document.title || '',
+        w: localWindow.innerWidth,
+        h: localWindow.innerHeight,
+        userConsent: JSON.stringify({
           // case of undefined, stringify will remove param
-          gdprApplies:
-            bidderRequest && bidderRequest.gdprConsent
-              ? bidderRequest.gdprConsent.gdprApplies
-              : '',
-          cmp:
-            bidderRequest && bidderRequest.gdprConsent
-              ? bidderRequest.gdprConsent.consentString
-              : '',
+          gdprApplies: utils.deepAccess(bidderRequest, 'gdprConsent.gdprApplies') || '',
+          cmp: utils.deepAccess(bidderRequest, 'gdprConsent.consentString') || ''
         }),
-      us_privacy:
-        bidderRequest && bidderRequest.uspConsent
-          ? bidderRequest.uspConsent
-          : '',
-    };
-
-    bidRequests.forEach(request => {
-      serverRequest.p.push(addPlacement(request));
-      const pubcid = getId(request, 'pubcid');
-      if (pubcid) {
-        serverRequest.pubcid = pubcid;
-      } else if (request.crumbs) {
-        if (request.crumbs.pubcid) {
+        us_privacy: utils.deepAccess(bidderRequest, 'uspConsent') || ''
+      };
+
+      bannerBidRequests.forEach(request => {
+        serverRequest.p.push(addPlacement(request));
+        const pubcid = getId(request, 'pubcid');
+        if (pubcid) {
+          serverRequest.pubcid = pubcid;
+        } else if (request.crumbs && request.crumbs.pubcid) {
           serverRequest.pubcid = request.crumbs.pubcid;
         }
-      }
-      const tdid = getId(request, 'tdid');
-      if (tdid) {
-        serverRequest.tdid = tdid;
-      }
-      const criteoId = getId(request, 'criteoId');
-      if (criteoId) {
-        serverRequest.cri_prebid = criteoId;
-      }
-      if (request.schain) {
-        serverRequest.schain =
-          JSON.stringify(request.schain);
-      }
-    });
-    serverRequest.p = '[' + serverRequest.p.toString() + ']';
-    return {
-      method: 'GET',
-      url: SERVER_ENDPOINT,
-      data: serverRequest
-    };
+        const tdid = getId(request, 'tdid');
+        if (tdid) {
+          serverRequest.tdid = tdid;
+        }
+        const criteoId = getId(request, 'criteoId');
+        if (criteoId) {
+          serverRequest.cri_prebid = criteoId;
+        }
+        if (request.schain) {
+          serverRequest.schain = JSON.stringify(request.schain);
+        }
+      });
+      serverRequest.p = '[' + serverRequest.p.toString() + ']';
+      serverRequests.push({
+        method: 'GET',
+        url: BANNER_SERVER_ENDPOINT,
+        data: serverRequest
+      });
+    }
+
+    if (videoBidRequests.length > 0) {
```

Comment:
do we need to make a similar check for banner requests? since now we are able to have just video requests?

Reply:
sure, we have it - see line 43:
`if (bannerBidRequests.length > 0) {`

Reply:
Whoops xD sorry bout that! Sounds good (: